actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ManageCourseBatchCount.java

```diff
@@ -65,31 +72,33 @@ public void execute(JobExecutionContext ctx) throws JobExecutionException {
         List<Map<String, Object>> listMap =
             (List<Map<String, Object>>) data.get(JsonKey.START_DATE);
         for (Map<String, Object> map : listMap) {
-          Map<String, Object> weakMap = new WeakHashMap<>();
-          weakMap.put(JsonKey.ID, (String) map.get(JsonKey.ID));
-          weakMap.put(JsonKey.COURSE_ID, (String) map.get(JsonKey.COURSE_ID));
-          weakMap.put(JsonKey.ENROLLMENT_TYPE, (String) map.get(JsonKey.ENROLLMENT_TYPE));
-          weakMap.put(JsonKey.COUNTER_INCREMENT_STATUS, true);
-          weakMap.put(JsonKey.COUNT_INCREMENT_DATE, ProjectUtil.getFormattedDate());
-          weakMap.put(JsonKey.STATUS, ProjectUtil.ProgressStatus.STARTED.getValue());
-          weakMap.put(JsonKey.UPDATED_DATE, ProjectUtil.getFormattedDate());
-          weakMap.put(JsonKey.HASHTAGID, (String) map.get(JsonKey.HASHTAGID));
-          CourseBatchSchedulerUtil.updateCourseBatchDbStatus(weakMap, true);
+          updateBatchMap(map);
         }
+        getContentAndStore(openBatchMap);
+        getContentAndStore(privateBatchMap);
+        contentMapUpdate(true);
+        openBatchMap.clear();
+        privateBatchMap.clear();
       }
       if (null != data.get(JsonKey.END_DATE)) {
         List<Map<String, Object>> listMap = (List<Map<String, Object>>) data.get(JsonKey.END_DATE);
         for (Map<String, Object> map : listMap) {
-          Map<String, Object> weakMap = new WeakHashMap<>();
-          weakMap.put(JsonKey.ID, (String) map.get(JsonKey.ID));
-          weakMap.put(JsonKey.ENROLLMENT_TYPE, (String) map.get(JsonKey.ENROLLMENT_TYPE));
-          weakMap.put(JsonKey.COURSE_ID, (String) map.get(JsonKey.COURSE_ID));
-          weakMap.put(JsonKey.COUNTER_DECREMENT_STATUS, true);
-          weakMap.put(JsonKey.COUNT_DECREMENT_DATE, ProjectUtil.getFormattedDate());
-          weakMap.put(JsonKey.STATUS, ProjectUtil.ProgressStatus.COMPLETED.getValue());
-          weakMap.put(JsonKey.UPDATED_DATE, ProjectUtil.getFormattedDate());
-          weakMap.put(JsonKey.HASHTAGID, (String) map.get(JsonKey.HASHTAGID));
-          CourseBatchSchedulerUtil.updateCourseBatchDbStatus(weakMap, false);
+          updateBatchMap(map);
+        }
+      }
```

Comment:
above 10 lines are repeated twice - once for start-date & once for end-date
Cn be made common - and processed 1-by-1 invoking the function call -> with passing arguments for variability